nixos/modules/services/networking/StevenBlack-hosts.nix

```diff
@@ -0,0 +1,49 @@
+{config, lib, pkgs, ...}:
+with lib;
+{
+  options.services.stevenBlackHosts = {
+    enable = mkEnableOption "filtering hosts via StevenBlack's hosts file";
+    flags = mkOption {
+      description = ''
+        Flags to passs while building the hosts file.
+        Note that compress and skipstatichosts are passed by default.
+        See also <link xlink:href="https://github.com/StevenBlack/hosts#command-line-options>here</link> for details.
+        Note that options that modify directly the hosts file or the global DNS cache will error out
+        as the host file is first built in the standard nix sandbox.
+      '';
+      default = {};
+      #TODO: cli setting type
+      type = with types; attrsOf (
+        oneOf [bool str (listOf bool) (listOf str)]
+      );
+
+    };
+    allowList = mkOption {
+      description = "List of hosts names to allow,one host per line.";
+      default = "";
+      example = ''
+        site.to.allow.example
+        another.to.allow.example.com
+      '';
+      type = types.lines;
+
+    };
+  };
+
+  config =
+    let
+      cfg = config.services.stevenBlackHosts;
+    in mkIf cfg.enable {
+
+      services.stevenBlackHosts.flags={
+        compress = mkDefault true;
+        skipstatichosts = mkDefault true;
```

Comment:
```suggestion
        skipStaticHosts = mkDefault true;
```

Reply:
No, as `flags` goes straight to the commandline flags of the build program (a bit like settings options in other modules).